openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -885,6 +895,30 @@ def edxnotes_visibility(self):
         """
         return self._original_course.edxnotes_visibility
 
+    @staticmethod
+    def validate_certificate_settings(course):
+        """
+        Take a course and update it's certificate display settings so they're valid pairings.
+        Updates passed in object.
+
+        Arguments:
+            course (CourseBlock): any course descriptor object
+
+        Returns:
+            None
+        """
+        # Backwards compatibility for existing courses that set availability date, didn't set behavior,
+        # and expect availability date to be used
+        if course.certificates_display_behavior == "" and course.certificate_available_date:
+            course.certificates_display_behavior = CertificatesDisplayBehaviors.END_WITH_DATE
+
+        if not CertificatesDisplayBehaviors.includes_value(course.certificates_display_behavior):
+            course.certificates_display_behavior = CertificatesDisplayBehaviors.END
+
+        # Null the date if it's not going to be used
+        if course.certificates_display_behavior != CertificatesDisplayBehaviors.END_WITH_DATE:
+            course.certificate_available_date = None
```

Comment:
I suspect that you might run into weird issues here, by trying to have CourseOverviews write back to the course in Studio. A couple of thoughts:

1. CourseOverviews are used in both Studio and LMS processes, so it's possible that you'll get a combination of draft and publish branches here–so it might push these changes to the published branch directly, instead of the draft branch that auto-publishes.
2. Doing this modification to the course content may (?) trigger another course publish (because it's changed), but I'm not sure if this code will actually persist to the modulestore at all.

In any event, while the [Studio/LMS boundaries ADR])https://github.com/edx/edx-platform/blob/master/docs/decisions/0005-studio-lms-subdomain-boundaries.rst) doesn't call out CourseOverviews explicitly, I think that this falls under the general directive that data flow should be unidirectional. It's fine to correct these values as they will be read from `CourseOverview`, but I don't think they should be persisted back to the modulestore from this layer.

Reply:
Ah, sorry I wasn't attempting to write back to modulestore at all. I was just attempting to edit the in-memory instance of the course that course overview was copying from. I'm gonna refactor this back to what it was where it's explicitly only reading from `course` and update the `course_overview` with the values. I _think_ that will clear up all three of the issues. 

Honestly, now that I'm looking at it again, this was a confusing way to code it. My bad.

Reply:
Updated. Lemme know if that looks better / alleviates the concerns. :)